File: TriviaClient/infrastructure/Request.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TriviaClient.Utils;

namespace TriviaClient.infrastructure
{
    class Request
    {
        private RequestID _id;
        private byte[] _buffer;

        public Request(RequestID id, byte[] buffer)
        {
            _id = id;
            _buffer = buffer;
        }

        public RequestID GetID()
        {
            return _id;
        }

        public byte[] GetBuffer()
        {
            return _buffer;
        }

    }
}

File: Trivia/Trivia/Trivia/Request.cpp
#include "Request.h"

Request::Request(RequestId id, std::time_t receivalTime, buffer buffer) : _id(id), _receivalTime(receivalTime), _buffer(buffer)
{
}

RequestId Request::getID()
{
	return _id;
}

std::time_t Request::getRecivalTime()
{
	return _receivalTime;
}

buffer Request::getBuffer()
{
	return _buffer;
}
